tests/integration/ardrive.int.test.ts

```diff
@@ -132,150 +135,281 @@ describe('ArDrive class - integrated', () => {
+			};
 			it('throws an error if its being moved inside any of its children folders', async () => {
+				stub(arfsDao, 'getPublicFolder').resolves(folderHierarchy.rootFolder);
+				stub(arDrive, 'getDriveIdAndAssertDrive').resolves(stubEntityID);
+				stub(arfsDao, 'getPublicChildrenFolderIds').resolves([
+					folderHierarchy.parentFolder.entityId,
+					folderHierarchy.childFolder.entityId,
+					folderHierarchy.grandChildFolder.entityId
+				]);
+
 				await expectAsyncErrorThrow({
 					promiseToError: arDrive.movePublicFolder({
-						folderId: stubEntityID,
-						newParentFolderId: stubEntityIDAltTwo // Alt ID Two will be in Children folder ids
+						folderId: folderHierarchy.rootFolder.entityId,
+						newParentFolderId: folderHierarchy.grandChildFolder.entityId
 					}),
 					errorMessage: 'Parent folder cannot be moved inside any of its children folders!'
 				});
 			});
 
 			it('throws an error if the new parent folder id matches its current parent folder id', async () => {
+				stub(arfsDao, 'getPublicFolder').resolves(folderHierarchy.childFolder);
+				stub(arDrive, 'getDriveIdAndAssertDrive').resolves(stubEntityID);
+				stub(arfsDao, 'getPublicChildrenFolderIds').resolves([folderHierarchy.grandChildFolder.entityId]);
+
+				await expectAsyncErrorThrow({
+					promiseToError: arDrive.movePublicFolder({
+						folderId: folderHierarchy.childFolder.entityId,
+						newParentFolderId: folderHierarchy.parentFolder.entityId
+					}),
+					errorMessage: `Folder already has parent folder with ID: ${folderHierarchy.parentFolder.entityId}`
+				});
+			});
+
+			it('throws an error if the new parent folder id matches its own folder id', async () => {
+				stub(arfsDao, 'getPublicFolder').resolves(folderHierarchy.parentFolder);
+				stub(arDrive, 'getDriveIdAndAssertDrive').resolves(stubEntityID);
+				stub(arfsDao, 'getPublicChildrenFolderIds').resolves([
+					folderHierarchy.childFolder.entityId,
+					folderHierarchy.grandChildFolder.entityId
+				]);
+
+				await expectAsyncErrorThrow({
+					promiseToError: arDrive.movePublicFolder({
+						folderId: folderHierarchy.parentFolder.entityId,
+						newParentFolderId: folderHierarchy.parentFolder.entityId
+					}),
+					errorMessage: 'Folders cannot be moved into itself!'
+				});
+			});
+
+			it('throws an error if the folder is being moved to a different drive', async () => {
+				stub(arfsDao, 'getPublicFolder').resolves(folderHierarchy.parentFolder);
+				stub(arfsDao, 'getPublicChildrenFolderIds').resolves([
+					folderHierarchy.childFolder.entityId,
+					folderHierarchy.grandChildFolder.entityId
+				]);
+
+				stub(arDrive, 'getDriveIdAndAssertDrive').resolves(stubEntityIDAlt); // Returns different Drive ID
+
 				await expectAsyncErrorThrow({
 					promiseToError: arDrive.movePublicFolder({
-						folderId: stubEntityID,
-						newParentFolderId: stubEntityID // Stub ID will MATCH current parentFolderId field
+						folderId: folderHierarchy.childFolder.entityId,
```

Comment:
These id choices are ok, but it confuses the test a bit because the focus should be on the driveId aspect. Here my eyes are drawn to the fact that you're moving the child folder to the parent folder that it's already in.

Reply:
d1d65e0